src/components/LmsConfigurations/SuccessFactorsConfigForm.jsx

```diff
@@ -0,0 +1,374 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import NewRelicService from '../../data/services/NewRelicService';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+
+export const REQUIRED_SUCCESS_FACTOR_CONFIG_FIELDS = [
+  'sapsfBaseUrl',
+  'sapsfCompanyId',
+  'sapsfUserId',
+  'key',
+  'secret',
+  'userType',
+];
+
+class SuccessFactorsIntegrationConfigForm extends React.Component {
+  state = {
+    invalidFields: {},
+    submitState: SUBMIT_STATES.DEFAULT,
+    active: this.props.config?.active,
+    error: null,
+    transmitTotalHours: this.props.config?.transmitTotalHours,
+  }
+
+  handleErrors = (error) => {
+    const errorMsg = error.message || error.response?.status === 500
+      ? error.message : JSON.stringify(error.response.data);
+    NewRelicService.logAPIErrorResponse(errorMsg);
+    return errorMsg;
+  }
+
+  /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createSuccessFactorsConfig = async (formData) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', this.props.enterpriseId);
+    try {
+      const response = await LmsApiService.postNewSuccessFactorsConfig(transformedData);
+      this.setState({ config: response.data });
+      return undefined;
```

Comment:
The linter made you do this? I'd prefer to return the previous line, I guess?

Reply:
"return the previous line" ? 
Can you return setting state?

And yes the linter made me do this. It didn't like `return` nor did it like not having a return at all.

Reply:
yea, `this.setState` probably also returns undefined, but that seems nicer to me somehow? Not critical, though.